### ISSUE TITLE
encoding error `Encoding::UndefinedConversionError`

### ISSUE THREAD

**Issue by ChrisCPO** [raen79/fie#27]
I'm getting an encoding error when publishing pools.

```ruby 
class BarsCommander < Fie::Commander
  pool :chat do
    state.bar = @published_object
  end

  def change_message
    @bar = Bar.find(1)
    @bar.message = Faker::Name.first_name
    @bar.save
    Fie::Pools.publish(:chat, @bar)
  end
end
```

View
```haml
%p= @bar.message

%div
  %button{"fie-click" => "change_message"}= "Change Message"
```
Error
`Encoding::UndefinedConversionError: "\x80" from ASCII-8BIT to UTF-8`

server logs 
```
[ActionCable] Broadcasting to pool_chat: {:command=>"publish_to_pool", :parameters=>{:subject=>:chat, :object=>"\x04\bo:\bBar\x16:\x10@attributeso:\x1EActiveModel::AttributeSet\x06;\x06{\tI\"\fmessage\x06:\x06ETo:)ActiveModel::Attribute::FromDatabase\t:\n@name@\b:\x1C@value_before_type_castI\"\rWendolyn\x06;\bT:\n@typeo:\x1EActiveModel::Type::String\b:\x0F@precision0:\v@scale0:\v@limit0:\x18@original_attribute0I\"\aid\x06;\bTo;\t\t;\nI\"\aid\x06;\bT;\vi\x06;\fo:\x1FActiveModel::Type::Integer\t;\x0E0;\x0F0;\x10i\r:\v@rangeo:\nRange\b:\texclT:\nbeginl-\t\x00\x00\x00\x00\x00\x00\x00\x80:\bendl+\t\x00\x00\x00\x00\x00\x00\x00\x80;\x110I\"\x0Fcreated_at\x06;\bTo;\t\t;\n@\x13;\vIu:\tTime\r\xA3\xC6\x1D\xC0C{+\xD3\x06:\tzoneI\"\bUTC\x06;\bF;\fU:JActiveRecord::AttributeMethods::TimeZoneConversion::TimeZoneConverter[\t:\v__v2__[\x00[\x00o:@ActiveRecord::ConnectionAdapters::PostgreSQL::OID::DateTime\b;\x0E0;\x0F0;\x100;\x110I\"\x0Fupdated_at\x06;\bTo;\t\t;\n@\x1C;\vIu;\x18\r\xA4\xC6\x1D\xC0X%{/\t;\x19I\"\bUTC\x06;\bF:\rnano_numi\x16:\rnano_deni\x06:\rsubmicro\"\a\x01p;\fU;\x1A[\t;\e[\x00[\x00@\e;\x110:\x17@aggregation_cache{\x00:\x17@association_cache{\x00:\x0E@readonlyF:\x0F@destroyedF:\x1C@marked_for_destructionF:\x1E@destroyed_by_association0:\x10@new_recordF:\x1E@_start_transaction_state{\x00:\x17@transaction_stateo:7ActiveRecord::ConnectionAdapters::TransactionState\a:\v@state:\x14fully_committed:\x0E@children[\x00:$@_new_record_before_last_commitF:\x18@validation_context0:\f@errorsU:\x18ActiveModel::Errors[\b@\x00{\x00{\x00:\x1D@mutations_from_database0:\x1E@_trigger_update_callbackT: @mutations_before_last_saveo:*ActiveModel::AttributeMutationTracker\a;\x06o;\a\x06;\x06U:#ActiveModel::LazyAttributeHash[\n}\tI\"\aid\x06;\bT@\x0FI\"\fmessage\x06;\bT@\vI\"\x0Fcreated_at\x06;\bT@\x17I\"\x0Fupdated_at\x06;\bT@!o:\x1DActiveModel::Type::Value\b;\x0E0;\x0F0;\x100{\tI\"\aid\x06;\bTi\x06I\"\fmessage\x06;\bTI\"\rPatricia\x06;\bT@\x13I\"\x1F2019-02-21 03:52:50.752451\x06;\bT@\x1CI\"\x1F2019-02-21 04:11:51.166414\x06;\bT{\x00{\x06@3o;\t\t;\n@3;\v0;\f@\x0F;\x110{\t@\bo:%ActiveModel::Attribute::FromUser\n;\n@\b;\vI\"\rWendolyn\x06;\bT;\f@\v;\x11o;\t\t;\n@\b;\v@;;\f@\v;\x110:\v@value@\n@\fo;\t\n;\n@\x0E;\vi\x06;\f@\x0F;\x110;8i\x06@\x13o;\t\n;\n@\x13;\v@<;\f@\x17;\x110;8U: ActiveSupport::TimeWithZone[\b@\x16I\"\bUTC\x06;\bT@\x16@\x1Co;7\n;\n@\x1C;\v@ ;\f@!;\x11o;\t\t;\n@\x1C;\v@=;\f@!;\x110;8U;9[\b@ @I@ :\x14@forced_changeso:\bSet\x06:\n@hash}\x00F:\"@attributes_changed_by_setterC:-ActiveSupport::HashWithIndifferentAccess{\x00", :sender_uuid=>nil}}
Could not execute command from ({"command"=>"message", "identifier"=>"{\"channel\":\"BarsCommander\",\"identifier\":\"bb4721f9-8a22-41cf-a75a-daa6b65ba0c7\"}", "data"=>"{\"caller\":{\"id\":\"\",\"class\":\"\",\"value\":\"\"},\"controller_name\":\"bars\",\"action_name\":\"show\",\"action\":\"change_message\"}"}) [Encoding::UndefinedConversionError - "\x80" from ASCII-8BIT to UTF-8]: /home/chris/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/activesupport-5.2.2/lib/active_support/core_ext/object/json.rb:38:in `encode' | /home/chris/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/activesupport-5.2.2/lib/active_support/core_ext/object/json.rb:38:in `to_json' | /home/chris/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/activesupport-5.2.2/lib/active_support/core_ext/object/json.rb:38:in `to_json' | /home/chris/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/activesupport-5.2.2/lib/active_support/json/encoding.rb:57:in `to_json' | /home/chris/.rbenv/versions/2.5.1/lib/ruby/2.5.0/json/common.rb:224:in `generate'
```

### Versions 
Rails 5.2.2
ruby 2.5.1p57

**Comment by ChrisCPO** [comment]
@raen79 This is happening here:

https://github.com/raen79/fie/blob/5cf7d4f53e0dab915cc77d6731bb28d2bf55e800/lib/fie/pools.rb#L22-L29

**Comment by raen79** [comment]
Hey could you potentially give me the structure of your `Bar` object? It is something within it's structure specifically that is preventing the marshalling.

**Comment by ChrisCPO** [comment]
I can but I noticed I have been able to replicate this on several different rails applications, it is possible it is something in `ActiveRecord`?

it is a pretty basic rails 5.2.3 model.

A user models dump
```"\x04\bo:\tUser\x19:\x10@attributeso:\x1EActiveModel::AttributeSet\x06;\x06U:#ActiveModel::LazyAttributeHash[\n}\x0EI\"\aid\x06:\x06ETo:\x1FActiveModel::Type::Integer\t:\x0F@precision0:\v@scale0:\v@limiti\t:\v@rangeo:\nRange\b:\texclT:\nbeginl-\a\x00\x00\x00\x80:\bendl+\a\x00\x00\x00\x80I\"\nemail\x06;\tTo:\x1EActiveModel::Type::String\b;\v0;\f0;\r0I\"\x17encrypted_password\x06;\tTo;\x13\b;\v0;\f0;\ri\x01\x80I\"\x17confirmation_token\x06;\tT@\x12I\"\x13remember_token\x06;\tT@\x12I\"\x0Ffirst_name\x06;\tT@\x10I\"\x0Elast_name\x06;\tT@\x10I\"\ntoken\x06;\tT@\x10I\"\x14connection_uuid\x06;\tT@\x10o:\x1DActiveModel::Type::Value\b;\v0;\f0;\r0{\x0EI\"\aid\x06;\tTi\aI\"\nemail\x06;\tTI\"\x14foobar@bazz.com\x06;\tTI\"\x17encrypted_password\x06;\tTI\"A$2a$10$f.dJwn7CnJpAvzwTuT/GiO54sAp//tXJTgkEPI4DmAtDdSKWGpetK\x06;\tTI\"\x17confirmation_token\x06;\tT0I\"\x13remember_token\x06;\tTI\"-879f5066247986e69d4095bc3d50565796f8ed6e\x06;\tTI\"\x0Ffirst_name\x06;\tTI\"\vChrisp\x06;\tTI\"\x0Elast_name\x06;\tTI\"\vbaccon\x06;\tTI\"\ntoken\x06;\tT0I\"\x14connection_uuid\x06;\tTI\"\x00\x06;\tT{\x00{\x06@\no:)ActiveModel::Attribute::FromDatabase\t:\n@name@\n:\x1C@value_before_type_cast0:\n@type@\v:\x18@original_attribute0{\x0E@\no;\x15\n;\x16@\n;\x17i\a;\x18@\v;\x190:\v@valuei\a@\x0Fo;\x15\n;\x16@\x0F;\x17@\x1D;\x18@\x10;\x190;\x1AI\"\x14foobar@bazz.com\x06;\tT@\x11o;\x15\n;\x16@\x11;\x17@\x1F;\x18@\x12;\x190;\x1AI\"A$2a$10$f.dJwn7CnJpAvzwTuT/GiO54sAp//tXJTgkEPI4DmAtDdSKWGpetK\x06;\tT@\x13o;\x15\n;\x16@\x13;\x170;\x18@\x12;\x190;\x1A0@\x14o;\x15\n;\x16@\x14;\x17@\";\x18@\x12;\x190;\x1AI\"-879f5066247986e69d4095bc3d50565796f8ed6e\x06;\tT@\x15o;\x15\n;\x16@\x15;\x17@$;\x18@\x10;\x190;\x1AI\"\vChrisp\x06;\tT@\x16o;\x15\n;\x16@\x16;\x17@&;\x18@\x10;\x190;\x1AI\"\vbaccon\x06;\tT@\x17o;\x15\n;\x16@\x17;\x170;\x18@\x10;\x190;\x1A0@\x18o;\x15\n;\x16@\x18;\x17@);\x18@\x10;\x190;\x1AI\"\x00\x06;\tT:\x17@aggregation_cache{\x00:\x17@association_cache{\x00:\x0E@readonlyF:\x0F@destroyedF:\x1C@marked_for_destructionF:\x1E@destroyed_by_association0:\x10@new_recordF:\x1E@_start_transaction_state{\x00:\x17@transaction_stateo:7ActiveRecord::ConnectionAdapters::TransactionState\a:\v@state:\x14fully_committed:\x0E@children[\x00:$@_new_record_before_last_commitF:\x18@validation_context0:\f@errorsU:\x18ActiveModel::Errors[\b@\x00{\x00{\x00:\x1D@mutations_from_database0:\"@attributes_changed_by_setterC:-ActiveSupport::HashWithIndifferentAccess{\x00:\x15@_already_called{\t:5validate_associated_records_for_game_membershipsF:*validate_associated_records_for_gamesF:5autosave_associated_records_for_game_membershipsF:*autosave_associated_records_for_gamesF:\x1C@new_record_before_saveF:\x1E@_trigger_update_callbackT: @mutations_before_last_save0:\x18@previously_changedC;.{\x00"
```

**Comment by raen79** [comment]
Hmm.. `fie` used to have ActiveRecord issues during the marshalling but they were fixed. Is there anything in common between these models?

**Comment by raen79** [comment]
Ah OK I figured it out, it has to do with ASCII characters not being converted properly (I think). I'll make a fix for it tonight.

**Comment by ChrisCPO** [comment]
oh wow, weird I tried 3 different rails apps on two different machines. euro sign odd. 

No worries jw how you figured it out.

**Comment by raen79** [comment]
I just realised it wasn't able to convert a single character from the log output you gave me and that character is a euro sign. So I realised it had to be that special characters were not being processed correctly. I tested this by putting other special characters all of which failed (eg. é, ë, etc...). Really sorry I haven't been able to get to it but it will be done today. Also, what are you using fie for? (just curious).
